library/src/main/java/com/vinted/coper/CoperImpl.kt

```diff
@@ -59,6 +72,52 @@ internal class CoperImpl(private val fragmentManager: FragmentManager) : Coper {
         }
     }
 
+    internal suspend fun getFragmentSafely(): CoperFragment {
+        val deferred = CompletableDeferred<CoperFragment>()
+        val fragment = getAttachedFragment()
+        fragment.lifecycle.addObserver(object : DefaultLifecycleObserver {
+            override fun onCreate(owner: LifecycleOwner) {
+                deferred.complete(fragment)
```

Comment:
Also, you are leaking `CoperImpl` inside `lifecycle` as the observer. You should remove the listener after you do not need it anymore.

Reply:
My recommendation is to write (or maybe find) extension function something like:
```
suspend fun Lifecycle.waitOnCreate()
```